src/lib/dma-trace.c

```diff
@@ -365,34 +365,89 @@ void dma_trace_flush(void *t)
 	dcache_writeback_invalidate_region((void *)t, size);
 }
 
-static void dtrace_add_event(const char *e, uint32_t length)
+static int dtrace_calc_buf_overflow(struct dma_trace_buf *buffer,
+	uint32_t length)
 {
-	struct dma_trace_buf *buffer = &trace_data->dmatb;
-	int margin;
+	uint32_t margin;
+	uint32_t overflow_margin;
+	uint32_t overflow = 0;
 
 	margin = buffer->end_addr - buffer->w_ptr;
 
-	/* check for buffer wrap */
-	if (margin > length) {
-		/* no wrap */
-		memcpy(buffer->w_ptr, e, length);
-		dcache_writeback_invalidate_region(buffer->w_ptr, length);
-		buffer->w_ptr += length;
-	} else {
+	/* overflow calculating */
+	if (buffer->w_ptr < buffer->r_ptr)
+		overflow_margin = buffer->r_ptr - buffer->w_ptr;
+	else
+		overflow_margin = margin + buffer->r_ptr - buffer->addr;
+
+	if (overflow_margin < length)
+		overflow = length - overflow_margin;
 
-		/* data is bigger than remaining margin so we wrap */
-		memcpy(buffer->w_ptr, e, margin);
-		dcache_writeback_invalidate_region(buffer->w_ptr, margin);
-		buffer->w_ptr = buffer->addr;
+	return overflow;
+}
 
-		memcpy(buffer->w_ptr, e + margin, length - margin);
-		dcache_writeback_invalidate_region(buffer->w_ptr,
-						   length - margin);
-		buffer->w_ptr += length - margin;
+static void dtrace_add_event(const char *e, uint32_t length)
+{
+	struct dma_trace_buf *buffer = &trace_data->dmatb;
+	uint32_t margin;
+	uint32_t overflow = 0;
+
+	/* amount of dropped entries */
+	static uint32_t dropped_entries;
```

Comment:
static variables should be declared at the top of the file.

Reply:
@lgirdwood Bartek is off today. I've patched his PR.

Reply:
@lgirdwood Bartek is off today. I've patched his PR.

Reply:
@lgirdwood Bartek is off today. I've patched his PR.

Reply:
@lgirdwood CI was failed because of to small .text section on CNL. Now it is ok.

Reply:
@bkokoszx can you also fix the CI checkpatch errors below.

Reply:
@bkokoszx I think re-pushing should restart CI. Travis can also be restarted manually by clicking on the link, not sure if the others can be done manually though.

Reply:
@lgirdwood Thanks, but it has been triggered already.